driver/src/main/scala/core/actors.scala

```diff
@@ -1245,29 +1245,13 @@ trait MongoDBSystem extends Actor {
 }
 
 @deprecated("Internal class: will be made private", "0.11.14")
-final class LegacyDBSystem private[reactivemongo] (
```

Comment:
I prefer not to mix deprecation cleanup and feature in the same PR. For now `LegacyDBSystem` must be kept as-is for compatibility.